### PR TITLE
Externals: Update WxWidgets to latest version from github,

### PR DESCRIPTION
**Pull request by phire**
Updating to the latest version of WxWidgets fixes an [annoying "Stack Overflow" bug](https://bugs.dolphin-emu.org/issues/8996) which has become much more common recently (possibly due to upgrading to windows 10, or some other change in dolphin. Who knows, the stacktrace clearly shows it's WxWidgets' fault)

As a side effect of upgrading we get a few UI upgrades for free. (On windows, I have checked linux/mac, and most of the time they will use whatever version of 3.0+ comes with the distro)

Dialogs with progress bars now have the modern (windows vista and later) show/hide details expander. Oh and the progress bar is animated. 

wx2

The gamelist now has a hover effect, the selected highlight looks a lot more pretty and behaves correctly around images like the banner, flags and star ratings.

New on left, Old on right.
wx

Finally, new asserts caught a minor bug, which has already fixed by #3140

---
This change is [Reviewable](https://reviewable.io/reviews/dolphin-emu/dolphin/3142)

### CODE REVIEW

File: Externals/wxWidgets3/include/wx/bitmap.h [hunk 1]
```diff
@@ -305,13 +302,7 @@ wxBitmap::
...
 ConvertToDisabled(unsigned char brightness) const
 {
-    // XXX comex: scale support
-    wxImage disabledImage = ConvertToImage().ConvertToDisabled(brightness);
-    #ifdef __APPLE__
-    return wxBitmap(disabledImage, -1, GetScaleFactor());
-    #else
-    return disabledImage;
-    #endif
+    return ConvertToImage().ConvertToDisabled(brightness);
```
Comment (JosJuice) [inline Externals/wxWidgets3/include/wx/bitmap.h]: Is this change intentional? comex has done some wx hacks in commits like 693cee36 and 18abc333

Comment (phire) [inline Externals/wxWidgets3/include/wx/bitmap.h]: Someone should have pushed such changes upstream, or something

**Comment by Parlane** [comment]
@dolphin-emu-bot rebuild

File: Externals/wxWidgets3/include/wx/bitmap.h (hunk 1 above)
Comment (lioncash) [inline Externals/wxWidgets3/include/wx/bitmap.h]: If I recall, handling for 2x is supported much better in the master version (this scaling should be done in the wxcocoa specific code now). I can look into this if you'd like.

**Comment by lioncash** [comment]
<3

**Comment by Parlane** [comment]
[ 15%] Building CXX object Externals/wxWidgets3/CMakeFiles/wx.dir/src/aui/auibar.cpp.o
In file included from /Users/matt/Projects/phire/Externals/wxWidgets3/src/aui/auibar.cpp:20:
In file included from /Users/matt/Projects/phire/Externals/wxWidgets3/include/wx/wxprec.h:12:
In file included from /Users/matt/Projects/phire/Externals/wxWidgets3/include/wx/defs.h:20:
In file included from /Users/matt/Projects/phire/Externals/wxWidgets3/include/wx/platform.h:489:
/Users/matt/Projects/phire/Externals/wxWidgets3/include/wx/chkconf.h:430:9: error: "wxUSE_ADDREMOVECTRL must be defined, please read comment near the top of this file."
#       error "wxUSE_ADDREMOVECTRL must be defined, please read comment near the top of this file."
        ^
/Users/matt/Projects/phire/Externals/wxWidgets3/include/wx/chkconf.h:438:9: error: "wxUSE_ACTIVITYINDICATOR must be defined, please read comment near the top of this file."
#       error "wxUSE_ACTIVITYINDICATOR must be defined, please read comment near the top of this file."
        ^
2 errors generated.
make[2]: *** [Externals/wxWidgets3/CMakeFiles/wx.dir/src/aui/auibar.cpp.o] Error 1
make[1]: *** [Externals/wxWidgets3/CMakeFiles/wx.dir/all] Error 2

**Comment by shuffle2** [comment]
btw the modern progress UI is already in dolphin's current version of wxw, but i had to disable it for some reason. Might want to check if that's still an issue...

File: Externals/wxWidgets3/include/wx/bitmap.h (hunk 1 above)
Comment (shuffle2) [inline Externals/wxWidgets3/include/wx/bitmap.h]: We have local changes to wxw which can't be expected to be pushed upstream. We need to make sure that stuff is still applied...if applicable...

**Comment by shuffle2** [comment]
for reference, the progress ui bug was http://trac.wxwidgets.org/ticket/13933 might be good to see if it's actually fixed or not, since comex has reapplied the patch in 2013, so doesn't appear to be fixed 4 years ago. see 54c3b8d357b7d69349090295893cdee5e4950b3c

Most other patches we had to wxw were for high-dpi and quieting warnings.

another maybe interesting one to see if still applicable is a38821fc389ea418415ccf107aa74a33d11999e8

**Comment by phire** [comment]
The progress UI bug appears to be fixed, I couldn't reproduce on windows 10 with either our md5 sum dialog, or the game list scanning on launch. Could someone try on windows 7?

OpenGL's getting stuck on hitting escape when full screened is fixed.

**Comment by Pringo** [comment]
Is this obsoleted by #3737?

**Comment by Pringo** [comment]
This PR can probably be closed now that #3737 was merged.

**Pull request closed without merging**